### ISSUE TITLE
PropertySortOrder: Allow JSON file to be used for sort order

### ISSUE THREAD

**Issue by kristerkari** [brigade/scss-lint#505]
I'm using `PropertySortOrder` linter to validate property ordering for my existing scss. Since I don't want to re-order all the properties by hand, I'm using [csscomb.js](https://github.com/csscomb/csscomb.js/) to do it for me automatically. 

I'm using [one of the csscomb preset files](https://github.com/csscomb/csscomb.js/blob/master/config/zen.json), but my problem is that scss_lint and csscomb use totally different formats to specify property ordering.

My feature request is to add a new option to `PropertySortOrder` where I can have sort order in a JSON file and have css_lint read it. That way importing sort order rules from csscomb will be really easy (more or less copy/paste).

Of course in the best case for me, I could pass my `.csscomb.json` file to `PropertySortOrder` and it would read the ordering from there. 

Using JSON would also enable [scss_lints ordering files](https://github.com/brigade/scss-lint/tree/master/data/property-sort-orders) to be compatible / easily converted to use with csscomb.

**Comment by sds** [comment]
I don't think it makes sense for `scss-lint` to support an arbitrary format used by a third-party tool. That introduces an unnecessary coupling (any changes to that format will break `scss-lint` lint).

You can easily convert `csscomb`'s preset format to an `scss-lint`-compatible one with the following short script:

```ruby
require 'json'
require 'yaml'

json = JSON.parse(File.read('zen.json'))

puts({
  'linters' => {
    'PropertySortOrder' => {
      'order' => json['sort-order'].flatten,
    }
  }
}.to_yaml)
```

You can modify this to insert `nil` between the groups if you would like that. 

Converting from an `scss-lint` preset order to JSON:

```ruby
require 'json'

order = []
current = []

File.foreach('concentric.txt')
    .reject { |line| line.start_with?('#') }
    .each do |line|
  if line.strip.empty?
    if current.any?
      order << current
      current = []
    end
  else
    current << line.strip
  end
end

if current.any?
  order << current
end

puts JSON.pretty_generate(
  'sort-order' => order
)
```

If `csscomb` supported a JSON format that didn't have any keys (i.e. was just the array of arrays format under the `sort-order` key) I would be more open to considering this. Happy to hear your thoughts.

**Comment by kristerkari** [comment]
Thanks for the scripts! This is exactly one part of the problem I have. Since scss-lint is Ruby based, I'm not able to easily write scripts like you just did. Could you consider to include those scripts to the repo, readme, wiki, etc. so that those are available to someone else that has the same problem?

Also, csscomb is not just any tool because it can quite powerful when used together with scss-lint (first apply linting, then comb the code using shared settings). It is too bad that open source library authors often overlook the benefits that their tool can bring when used together with another library. For example, would be nice to see these two projects sharing their property order configurations (the order, not the format).

> I don't think it makes sense for scss-lint to support an arbitrary format used by a third-party tool. That introduces an unnecessary coupling (any changes to that format will break scss-lint).

Yes absolutely not, my request was mainly for supporting JSON in general, but having something that would support csscomb out of the box would have been a plus.

Supporting a third party format does not mean that you have to create coupling. IMO the correct way of implementing something like that would be a plugin system that would enable _any_ format to be supported. Look at [PostCSS](https://github.com/postcss/postcss#how-to-develop-postcss-plugin) for example, with its plugin based architecture it can support almost any kind of tool/format. But anyway, these were just my 2 cents on how to support something without creating coupling.

**Comment by sds** [comment]
Thanks for your thoughts, @kristerkari.

I'm happy to say that `scss-lint` supports plugins as of #440, so it sounds like the feature you are looking for is supportable if someone is willing to take on the responsibility of maintaining the plugin.

@brigade can't take on the task of maintaining all of these different integrations (we aren't experts with every toolset&mdash;no one is). Hopefully someone from the community will step up and and make those plugins a reality if the need for them is real.

**Comment by kristerkari** [comment]
Thanks! Sounds awesome.

> @brigade can't take on the task of maintaining all of these different integrations (we aren't experts with every toolset—no one is). Hopefully someone from the community will step up and and make those plugins a reality if the need for them is real.

If you again look at PostCSS as an example, you can see that it is successful because of good documentation, guidelines, api docs, faq and community. Also the guy behind PostCSS goes around and gives conference talks about PostCSS. It is not easy, but it is doable.
